### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by AiniusG**

### CODE REVIEW

File: backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        _controller = new MemoryGameWithNumbersController();
+    }
+
+    [Fact]
+    public void StartGame_ShouldReturnGridWithCorrectNumbersAndNulls()
+    {
+        var maxNumber = 5;
+        var result = _controller.StartGame(maxNumber);
+
+        if (result.Result is OkObjectResult okResult && okResult.Value is List<int?> gridNumbers)
```
Comment (Aidas21J) [inline backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs:29]: idk, probably geriau vietoj sito if'o ekvivalenciu assertu pridet. Kituose testuose, kur panasus dalykas, irgi.

File: backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        {
+            Assert.Equal(200, okResult.StatusCode);
```
Comment (Aidas21J) [inline backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs]: Tuos `200` geriau iskelt i constanta, ir visur ja naudot.

File: backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            Assert.NotNull(gridNumbers);
+            Assert.Equal(16, gridNumbers.Count);
```
Comment (Aidas21J) [inline backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs:33]: Ateity butu gerai, kad grid'as turetu jo ismatavimu kintamaji ar konstanta.

File: backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    [Fact]
+    public void StartGame_ShouldReturnGridWithRandomizedOrder()
+    {
+        var maxNumber = 5;
+
+        var result1 = _controller.StartGame(maxNumber);
+        var result2 = _controller.StartGame(maxNumber);
+
+        if (result1.Result is OkObjectResult okResult1 && result2.Result is OkObjectResult okResult2)
```
Comment (Aidas21J) [inline backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs:51]: Nzn ar blogai supratau, bet cia ir kituose testuose, jei `if`'e salyga evaluatin'as i `false`, tai testas, man artodo, turetu failint.

File: backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        _controller.StartGame(maxNumber);
+        var correctAttempt = Enumerable.Range(1, maxNumber).Select(n => (int?)n).ToList();
+
+        var result = _controller.CheckAttempt(correctAttempt);
+
+        if (result.Result is OkObjectResult okResult)
+        {
+            Assert.Equal(200, okResult.StatusCode);
+            Assert.True((bool)okResult.Value);
+        }
```
Comment (Aidas21J) [inline backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs]: Sitas patternas daug kur paskartoja. Might be a good idea extractint i `static` methoda.

File: backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    [Fact]
+    public void CheckAttempt_ShouldReturnBadRequest_WhenGameNotStarted()
+    {
+        var userAttempt = new List<int?> { 1, 2, 3, 4, 5 };
+
+        var result = _controller.CheckAttempt(userAttempt);
+
+        var badRequestResult = Assert.IsType<BadRequestObjectResult>(result.Result);
+        Assert.Equal(400, badRequestResult.StatusCode);
+        Assert.Equal("Game not started.", badRequestResult.Value);
```
Comment (Aidas21J) [inline backend/testing/MemoryGameWithNumbers/MemoryGameWithNumbersControllerTest.cs:69]: Skonio reikalas, bet bent jau man labiau patinka rasyt `Assert.Equal(variable, constant)`, o ne `Assert.Equal(constant, variable)`. Biski patogiau skaityt. Bet again, skonio reikalas, tai if you prefer this way, as long as you're consistent with it, everything is fine by me.

**Review by Aidas21J — APPROVED** [review]
Approved. Some nitpicks from me, bet everything looks good.

**Review by Dax0s — APPROVED** [review]

**Pull request merged**